### ISSUE TITLE
Nixie “Schedules” start/end timer settings not responding as expected

### ISSUE THREAD

**Issue by rock-crusher** [tagyoureit/nodejs-poolController#367]
Issue:  In the nixie scheduler when setting the on/off times the minutes are not changeable while in the 12:00pm hour. When you try, it automatically reverts back to 12:00pm. For instance, if you set the time as 12:37pm the system automatically resets the time back to12:00pm for some reason. All the other hours 1-11 am or pm you can set any minute you desire.  
I’m sure it’s just something I’m doing wrong.
Any Ideas?

![Schedules](https://user-images.githubusercontent.com/94593954/142744864-0923f9b3-88ac-4f77-8b67-9078e3bd408b.jpg)

**Comment by rstrouse** [comment]
I’ll have a look.  Perhaps we have a validation issue with the field.

I assume this means you have figured out all the hardware setup.

**Comment by rock-crusher** [comment]
Hi again rstrouse,
When I type it in, the system just automatically defaults it back to 12:00pm.
As far as getting it all figured out… lol, far from it. I’m just playing around with the relays right now on my work bench getting familiar with the circuits and scheduling in Nixie. I’m able to schedule the relays and get them to fire when I want, but then something glitches and things start going wonky. I’ll lose control of a relay and can’t get it to work again. I’ll refresh everything, reboot, delete the circuits/schedules start over and add them new again and nothing, I can’t get even one now to engage. Excuse me; I can get them all to work manually in REM by clicking on the relay buttons, but it’s like Nixie somehow starts acting erratic and disassociates itself. I can’t figure out the egg timer. Not sure quite how to initiate it, and when I think I do, it never shuts off no matter how the timing parameters are set. Another thing I haven’t figured out is the buttons in Nixie’s settings/appearance tab. I can click on them, but they do nothing I’m aware of.
Once again I must be doing something wrong, maybe you can throw me a bone or two.
Thank you rstrouse  

<img width="318" alt="Circuit #3" src="https://user-images.githubusercontent.com/94593954/142756424-db8d5f86-caac-4650-81de-8dd03a4e2f85.png">

<img width="386" alt="Nixie Home Page" src="https://user-images.githubusercontent.com/94593954/142756436-ee24deb0-4afd-43f1-8348-d783215cd58b.png">

<img width="500" alt="Relay #7" src="https://user-images.githubusercontent.com/94593954/142756440-161bb486-f4ed-433d-a939-80a0d603b5cc.png">

<img width="309" alt="appearance tab" src="https://user-images.githubusercontent.com/94593954/142756445-a1c7377f-f518-4c52-8f39-15eb46f7d0fc.png">

**Comment by rstrouse** [comment]
**Issue 1**
I pushed up a fix for the time selection weirdness.  There was a bug in the handling for the 12 hour clock were it was not honoring the minutes.

**Issue 2**
For your relay, lets make sure we are looking at the same thing.   Are you saying that when you click on circuit 3 in dashPanel it does not change the little indicator to green?  If it does then it is sending the signal to REM and REM is responding that it processed the request.

If you are looking at the little LEDs on the Sequent board, the Sequent 8 relay board has the screen printing backwards on the indicator LEDs on the board.  At the time of printing the numbering got printed on the other side of the board.  What this did is make the relay 1 light the relay 8 led and relay 7 light the relay 2 led.

For troubleshooting you can open up the `pm2 monit` then select the njsPC item on the monitor.  Then watch the console to see if there are any errors when you press circuit 3 on the dashPanel.

**Issue 3**
@tagyoureit pushed some changes to fix the egg timer.  If you want to know more about what an egg timer does it is in this section of the wiki
https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup#circuits----aux-circuits

**Comment by rstrouse** [comment]
The buttons in the last screenshot are to change the way dashPanel displays the sections of the display.  Simply drag them to the location and order where you want them to display.  For instance, in your case you do not have a filter pressure sensor so simply drag the filters bubble to the hidden items list.  I also always drag the virtual circuits bubble there as well.

**Comment by rock-crusher** [comment]
Hi again,
Well, I tried everything you suggested to no avail; the only thing that it seemed to fix is that I now have control of circuit #3 again. Problem still exists for the egg timer, it never turns off. The 12:00pm issue still exists too.
Funny thing with the egg timer is when the button is pushed it engages and the green light turns on firing the correct relay, but the countdown on the button starts out at 3hrs 50min even though I have it set for only 1min.
By the way, I have tried the above suggestions you have outlined twice with no success and yes I rebooted along with the hard refresh of the browser.

<img width="386" alt="Features cir #3" src="https://user-images.githubusercontent.com/94593954/142780725-ed81818d-f4f2-4ae5-bb3b-26c5c672c504.png">

<img width="323" alt="#3 egg timer" src="https://user-images.githubusercontent.com/94593954/142780735-d7d156bd-e6a3-4296-ad03-beb83cef2765.png">

![pool controller](https://user-images.githubusercontent.com/94593954/142780750-8aeac2f2-7104-4eb9-8dc2-61b82ca7e038.png)

![dash panel](https://user-images.githubusercontent.com/94593954/142780758-8357d7a4-62f0-4e9c-acfd-1515258e3def.png)

**Comment by rstrouse** [comment]
That is the correct behavior for the timer because you started it in the middle of the schedule so it will work off the schedule time.

**Comment by rock-crusher** [comment]
rstrouse, Lol… makes perfect sense! I feel like such a dumb ass. Also, the 12:00 issue is now working great!! (wonder why it took so long for it to take effect on my end?).  
Anyway, I’ll continue on with my quest. As you know this is all extremely new for me, but let me tell you a great learning experience and although frustrating at times (my dumb ass) it is super fun too!
Thanks again for taking the time to help me.

**Comment by rstrouse** [comment]
No worries happy to help.  Some of this stuff ends being obvious when you have the goggles that were required to write the functionality then when somebody not wearing them looks at it you realize where the gaps are in function and documentation.  

The reason the 12:00 issue finally fixed itself is likely because you didn't get a hard refresh on the browser and it was using the cached script.  These have an expiration date set on them so eventually the browser asked for the new files with the new script in it.  dashPanel is a dynamic set of pages that are built on the fly so the cache mechanisms on the express web server are not able to appropriately look for cached files.

I suspect that most folks would have difficulty with the wiring but muddle their way through the configuration.   I spent more time documenting (how to) on the wiring part but less time on configuration.  This is partially, because the concepts are similar to commercial offerings out there.  

Bear in mind there is definitely the possibility for software errors since the source of all errors is code and Nixie is made up mostly of code.  Add to that neither I or @tagyoureit have all the equipment we rely a lot on folks providing feedback on their equipment and take some educated guesses when we need to.

**Comment by rock-crusher** [comment]
Back again, as you knew I would be… lol.
Maybe a little help with getting my VSF pump setup? I’ve connected the RS485 cable to the pump and Building Automation Card, Green on A & yellow on B terminal. The jumpers on the card are in place, the RS485 pass through is in place. I keep getting the: Unable to open RS485 port message?
I know there is some procedure for this, so if you could just send me in the right direction.
Thank you

**Comment by rstrouse** [comment]
Did you change the settings on the pi as indicated here.
https://github.com/rstrouse/relayEquipmentManager/wiki/Sequent-Microsystems-Boards#rs-485-tab

If you did then navigate to the Controller -> RS485 tab in the dashPanel configuration screens.  Enter ‘/dev/ttyS0’ as the port name.

**Comment by rock-crusher** [comment]
Yes the pi has already been changed, and I keep getting the Unable to open RS485 port message.

![RS485 port](https://user-images.githubusercontent.com/94593954/142801678-69dde22f-3658-4726-9e82-d4f5d2383289.png)

![pi config](https://user-images.githubusercontent.com/94593954/142801946-783fcb65-282b-4022-a573-cbbda0dcb934.png)

**Comment by rstrouse** [comment]
First lets verify the following
1. The jumpers are installed on the MEGA-BAS
![image](https://user-images.githubusercontent.com/47839015/142886100-27c687ec-2a08-44fd-9de6-77415e7513f1.png)
2. The MEGA-BAS is configured for RS485 Pass thru
![image](https://user-images.githubusercontent.com/47839015/142886408-849beb62-349d-4dbb-b747-2759717066ec.png)
3. After you reboot the pi run the following command.  It should report that this device exists on the pi
`ls /dev/ttyS0`
![image](https://user-images.githubusercontent.com/47839015/142888316-0a587b19-0263-444f-843b-908887e7ed4d.png)

**Comment by rock-crusher** [comment]
Yes, it is set up exactly as you show it above.

**Comment by rock-crusher** [comment]
I have been configuring all of this via wifi. Could this have anything to do with why I keep receiving the Unable to open RS485 port /dev/ttyS0 ?

**Comment by rock-crusher** [comment]
Also, I don’t have the disc battery installed in the MEGA-BAS, is this a problem? Is it necessary?
The kit I bought didn’t have one included. What kind of battery is it when I go to purchase one?

**Comment by rstrouse** [comment]
First some answers.
WiFi should not matter.
The battery is a CR2450 but it is for the RTC on the board so it should not matter.

I have a couple of questions

Did you set up an RDP or VNC connection to access the pi?  
Is this a Pi3 or Pi4?
Just to be sure when you perform the command `ls /dev/ttyS0` it returns the part in yellow like above?

Things to try
Turn Bluetooth off on the pi (it will be the little Bluetooth icon on the pi taskbar)?

Run the command `ls -l /dev/ttyS0` and paste the results here.

Run the command `groups pi` and paste the results here.

If the port exists lets try to see if there is any traffic on the port.  Hit ctrl + c to exit.
`od -x < /dev/ttyS0`

**Comment by rock-crusher** [comment]
- _Did you set up an RDP or VNC connection to access the pi?_

Yes, RDP, but I’m not doing the set-up using that. I have a keyboard, mouse & monitor directly connected to the PI4.

- Is this a Pi3 or Pi4?

PI4

- _Just to be sure when you perform the command ls /dev/ttyS0 it returns the part in yellow like above?_

Yes it does, attached image of result.

- _Things to try
Turn Bluetooth off on the pi (it will be the little Bluetooth icon on the pi taskbar)?_

It has always been off, attached image.

- _Run the command ls -l /dev/ttyS0 and paste the results here._

Attached image

- _Run the command groups pi and paste the results here._

- _If the port exists lets try to see if there is any traffic on the port. Hit ctrl + c to exit._
_od -x < /dev/ttyS0_

Attached image

![1](https://user-images.githubusercontent.com/94593954/142932835-04f6ba8f-9600-4530-8b1d-fabc0526b4ec.png)

![2](https://user-images.githubusercontent.com/94593954/142932862-1559eb6b-6581-43bf-b623-bd188aef00fa.png)

![04](https://user-images.githubusercontent.com/94593954/142933731-91ee905c-6f3d-4df2-a928-8339c0157e56.png)

![3](https://user-images.githubusercontent.com/94593954/142932884-a49a2743-5759-4a87-aba7-a6099a7ad7fa.png)

**Comment by rock-crusher** [comment]
Does the pump (VSF) need to be configured in some special way from its menu, or is it just plug-n-play? 
I couldn’t find anything scrolling through its menu regarding this.

**Comment by rstrouse** [comment]
It will be just plug and play.  Nixie will send the appropriate commands via RS485.  Under the pumps section in the configuration you add the pump as a VSF and tell it which circuits/features should make the pump run at which speed.

**Comment by rock-crusher** [comment]
Yes, I’ve done all that and set-up a schedule for it as well. When the schedule dictates for it to turn on “Spa” as I have it set-up for, the spa button green light turns on in the "features" window and it also shows the amount of time left for it to run. Problem is, the actual pump does nothing.
Any other thoughts, suggestions?

**Comment by rstrouse** [comment]
This is because we have not been able to get your RS485 port working.  I am researching now why you might be getting a permissions issue opening the port.

When Nixie starts communicating with the pump the VSF display will change to "Display Not Active" and you will be controlling the pump through Nixie over RS485.

**Comment by rock-crusher** [comment]
Understood, 
Thank you for all your efforts!

**Comment by rock-crusher** [comment]
Funny thing now, when I check the RS485 port on the controller tab I now get this: attached image
As opposed to yesterday when I got this: attached image
It still doesn’t work, but can what is circled in red be changed to get it working? 

<img width="318" alt="01" src="https://user-images.githubusercontent.com/94593954/142959486-d4222252-2bd1-4871-9ef0-a7413a453882.png">

![RS485 port](https://user-images.githubusercontent.com/94593954/142959514-b9bae086-4e30-4c1b-93c5-d4a1cd463c7f.png)

**Comment by rstrouse** [comment]
It cannot change it if it cannot access it.

**Comment by rock-crusher** [comment]
My bad, just realized the port name was changed from Local to Network. I must have done it when screwing around.

**Comment by rstrouse** [comment]
No worries, I am trying to figure out how to change the port permissions so that they stick.  They really made this overly complicated when they added bluetooth to the pi.

**Comment by rstrouse** [comment]
Can I get you to enter the following command then try to save the port

`sudo chmod 0666 /dev/ttyS0`

If this works it will only work until the next reboot.  Not sure why your port would not have all the permissions.

**Comment by rock-crusher** [comment]
Just tried it, unfortunately it did not work 
![03](https://user-images.githubusercontent.com/94593954/143090686-d16ba732-b557-4643-bb02-73b1026f126f.png)

![02](https://user-images.githubusercontent.com/94593954/143090725-a45b5015-6b19-43f9-8c32-d713a62fe958.png)

**Comment by rstrouse** [comment]
Just for shucks can I get you to change the port name to `/dev/serial0` then try to save it.  It is so very odd that the permissions are not correct for this.  I have tried repeatedly to make a pi do what yours is doing but I cannot seem to get it to fail using `/dev/ttyS0` when bluetooth is disabled and the MEGA-BAS is set to pass thru on the RS485 tab.

**Comment by rock-crusher** [comment]
yes, it saved, now what?

**Comment by rstrouse** [comment]
Are you seeing traffic now?  Do you have the VSF pump defined in the configuration?

**Comment by rock-crusher** [comment]
No traffic, nothing has changed.
Yes, VSF pump is defined in the configuration.

**Comment by rstrouse** [comment]
Run `ls -l /dev/serial*` and post the results here.

**Comment by rock-crusher** [comment]
![09](https://user-images.githubusercontent.com/94593954/143124992-3038728e-9202-4b3c-a7a4-dd73362637c3.png)

**Comment by rstrouse** [comment]
Run the following commands
`cd ~/nodejs-poolController`
`git pull`

Wait until dashPanel reconnects then select the hamburger menu.  

Then navigate to the Logging tab and press Capture Replay then Begin Capture on the dialog.

After you do that go to the RS485 Port and save the port.

Go back to the Logging tab and press Cancel Capture.  It will download a file to your browser.  Upload that file here.

**Comment by rock-crusher** [comment]
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/7591164/replay.zip)

**Comment by rstrouse** [comment]
Ok so it is trying to send messages to the pump but the pump is not responding.  Flip the green and yellow wires going to the pump.

**Comment by rock-crusher** [comment]
OK, did that and still nothing has changed.

**Comment by tagyoureit** [comment]
Make sure the pump isn't running a schedule; try to put it in Filter 1 and try again.

**Comment by rock-crusher** [comment]
"try to put it in Filter 1 and try again."
Going to need help with that

**Comment by rock-crusher** [comment]
should I put the wires back, green on A and yellow B?

**Comment by tagyoureit** [comment]
On your pump, how has it been running in standalone mode?  On a schedule?

**Comment by rock-crusher** [comment]
no schedule

**Comment by rstrouse** [comment]
A should be yellow and B should be green.  

Lets try something different.  I am surprised that the pump does not say "Display Not Acitve"

`cd ~`
`git clone https://github.com/SequentMicrosystems/megabas-rpi.git`
`cd megabas-rpi`
`sudo make install`

Then after all this has finished run the following command and shoot  a screenshot.
`megabas 0 rs485rd`

Then issue the following command
`megabas 0 rs485wr 0 9600 1 0 1`

**Comment by rock-crusher** [comment]
![11](https://user-images.githubusercontent.com/94593954/143144710-159974cd-72b9-4a36-90ac-318d6accee42.png)

**Comment by rstrouse** [comment]
Hmmm.... that is exactly what I would have expected to see if it was all configured properly so nothing there.

I have installed this board 3 times and all 3 times the RS485 connection was on `/dev/ttyS0`.   There were not any issues with permissions.   Lets me look inside your `/boot/config.txt`.  You upload that file here from your raspberry pi?

**Comment by rock-crusher** [comment]
[config.txt](https://github.com/tagyoureit/nodejs-poolController/files/7591598/config.txt)

**Comment by rock-crusher** [comment]
I’m a little confused about the conflicting RS485 color code.
In the below image it says green to A, yellow to B, but you have me the opposite?

<img width="321" alt="Q" src="https://user-images.githubusercontent.com/94593954/143148302-c805cfc6-3615-4c15-adf5-391e48a14d7d.png">

**Comment by rstrouse** [comment]
Yes the original documentation that we got this information from was backwards.  We need to update the wiki.

**Comment by rock-crusher** [comment]
> Ok so it is trying to send messages to the pump but the pump is not responding. Flip the green and yellow wires going to the pump.

If it is sending messages to the pump, then could it be something wrong with my pump? 
It does work when controlled directly from the panel screen mounted on top of the motor itself.

**Comment by rstrouse** [comment]
Alright this is so confounding.  Something about the pi OS is not right and all the goofy recommendations on the web are brutally complicated.  We may be better off re-imaging the OS and starting over.

I do not think there is anything wrong with your pump.  While it is writing to the port I am not so sure that the message is actually making it to the pump.  All of the gymnastics that we have been performing wrt the port permissions makes me think that the messages make it to the transmit buffer they end up blowing out into the ether.

How do you feel about re-imaging the SD card and starting over?

**Comment by rock-crusher** [comment]
I'm ok with what ever you recommend, but will need your help getting it done.

**Comment by rstrouse** [comment]
So I assume you bought a pi kit with a pre-loaded SD card.  It should also have an adapter to put the SD card in with a USB port.  Pull the SD card out of the pi and insert it into this adapter.

So are you a Mac or a Windows user?

**Comment by rock-crusher** [comment]
Windows

**Comment by rstrouse** [comment]
Ok so follow these instructions to get the image installed on the Raspberry Pi.  Once you do that we need to hook the pi up to a mouse, keyboard, and monitor to install XRDP.

https://projects.raspberrypi.org/en/projects/raspberry-pi-setting-up/2

**Comment by rstrouse** [comment]
Once you get the pi back up and running with a keyboard, mouse, and monitor.  

Run the following commands so you can get access to it from your pc.  Set the basic settings for the pi and hook up a network connection to it.  Then run the following commands.

`sudo apt-get update`
`sudo apt-get upgrade`
`sudo apt-get install xrdp`

After you do this you should be able to access your pi using Remote Desktop Connection (rdp).  Open rdp and enter raspberrypi into the computer field and press connect.  Answer yes to the prompts and the login screen for the pi should appear.  You are now operating the pi headless so you can stop using the monitor, keyboard and mouse.
![image](https://user-images.githubusercontent.com/47839015/143157357-4fd2aa8f-2b29-4e42-aabc-e3dd87693249.png)

Next you will want to perform the install of the software starting with the pre-requisites.
https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#install-software-prerequisites

When you get to the part where you are selecting the interfaces.  Disable the Serial Console and enable the Serial Port.

Once you get to that point go ahead and configure the MEGA-BAS and your relay card and let me know when you get to that point.

**Comment by rock-crusher** [comment]
Ok, did all that, now just start everything all over again in DIY Standalone Nixie Pool Controller?

**Comment by rock-crusher** [comment]
sorry the page didn't refresh, disregard my above comment

**Comment by rock-crusher** [comment]
> Once you get the pi back up and running with a keyboard, mouse, and monitor.
> 
> Run the following commands so you can get access to it from your pc. Set the basic settings for the pi and hook up a network connection to it. Then run the following commands.
> 
> `sudo apt-get update` `sudo apt-get upgrade` `sudo apt-get install xrdp`
> 
> After you do this you should be able to access your pi using Remote Desktop Connection (rdp). Open rdp and enter raspberrypi into the computer field and press connect. Answer yes to the prompts and the login screen for the pi should appear. You are now operating the pi headless so you can stop using the monitor, keyboard and mouse. ![image](https://user-images.githubusercontent.com/47839015/143157357-4fd2aa8f-2b29-4e42-aabc-e3dd87693249.png)
> 
> Next you will want to perform the install of the software starting with the pre-requisites. https://github.com/tagyoureit/nodejs-poolController/wiki/DIY-Standalone-Nixie-Pool-Controller#install-software-prerequisites
> 
> When you get to the part where you are selecting the interfaces. Disable the Serial Console and enable the Serial Port.
> 
> Once you get to that point go ahead and configure the MEGA-BAS and your relay card and let me know when you finish that configuration.

Ok, I'll get started on this, thanks.

**Comment by tagyoureit** [comment]
I have the Mega-BAS card and have not yet enabled RS-485 pass thru.  I tried it and ran into the same issue.  Ironically, there seem to be thousands of similar issues out there but I figured out a quick and easy way to do this that I didn't see mentioned anywhere.  Give this a try and let me know if this works for you.

1. Run `sudo raspi-config`.
2. Select 3 `interface options`
3. Select P6 `serial port`
4. **Important one** - Would you like a login shell to be accessible over serial?  NO.
![image](https://user-images.githubusercontent.com/1646840/143178281-b9135671-81b8-4bbf-9f88-71116ba95e2c.png)
5. Would you like the serial port hardware to be enabled?  Yes.
![image](https://user-images.githubusercontent.com/1646840/143178316-cefb37fe-416d-4d8d-b515-7ca5924598ac.png)

Final settings should look like:
![image](https://user-images.githubusercontent.com/1646840/143178362-f94a021f-453a-4331-9792-c38d9d424811.png)

When I answered yes to the login shell (easy to do!) that is where this problem is caused.  I tried it about 3 times and was able to replicate the `Unable to open port` each time I answered yes to the login shell question.

**Comment by rock-crusher** [comment]
Hi tagyoureit,
unfortunately I haven't been able to complete getting everything installed.
I'm stuck once again, I can't seem to get PM2 installed and running.
When I try and run the command: sudo npm install -g pm2 I get this:
![l](https://user-images.githubusercontent.com/94593954/143181168-7a45c173-16b4-4396-8b79-1308e06529f8.png)

**Comment by rstrouse** [comment]
After installing node you must open a new console window or the path to num will not be found.

**Comment by rock-crusher** [comment]
> After installing node you must open a new console window or the path to num will not be found.

Sorry, I don't understand

**Comment by rock-crusher** [comment]
I didn't have a problem installing pm2 last time around, now I'm having trouble and don"t understand. 
"open a new console window"   ??

**Comment by tagyoureit** [comment]
He means close the terminal and re-open it.  That will start a new shell session.

**Comment by rock-crusher** [comment]
Thank for that info tagyoureit.
That is what I have been doing, or so I thought.
I'll try again

**Comment by rock-crusher** [comment]
Same thing, not working.

**Comment by rock-crusher** [comment]
I give-up for now

**Comment by tagyoureit** [comment]
It looks like you don't have node/npm installed yet.  These instructions will get you through it: https://linuxize.com/post/how-to-install-node-js-on-raspberry-pi/

On the following command in the article, replace the `10` with `12`:
`curl -sL https://deb.nodesource.com/setup_12.x | sudo bash -`

**Comment by rstrouse** [comment]
I think you may have missed the `nvm install 12.18.3` command.  This installs node and npm using the node version manager.

**Comment by rock-crusher** [comment]
Hi guys,
I'm giving this another try. I followed tagyoureit's suggestions above and it installed. 
So, got the pm2 done along with logrotate.
Question 1, I didn't have to do this first time around, all I did before was copy past from DIY Standalone Nixie Pool Controller see attached image "b" below and it worked fine, so why this time different?
Now, i'm having problem with REM. I got the Mega-Bas ok, but when trying to install the relay card the 3-0x03 unknown address is incorrect. 
It wont let me put in the correct address. When I click the + add device button and then click the add 12c address to put the correct address in, the window never opens, the program automatically chooses some generic one as shown in image "a" below which is wrong and will not work.
Question 2, Why doesn't the window open so I can put in the correct address like it did before in the first time around?
As per image "c" below, address 72 & 63 is what I want. It seems to get the 72 correct which I assigned for the Mega-Bas, but wrong for the other address (relay card) and I cant change it?

Anyway I'm stuck here and need help.

![b](https://user-images.githubusercontent.com/94593954/143324344-f77636bc-1076-4222-bee2-cd7cb0e929a3.png)

![a](https://user-images.githubusercontent.com/94593954/143324374-d1c46223-0eb3-4ca6-82c5-b1c47f0a0d3b.png)

![c](https://user-images.githubusercontent.com/94593954/143324397-a13469df-762e-4d40-af4a-47ca5b096afb.png)

**Comment by rstrouse** [comment]
The detection method in REM is passive.  It does this so that it does not interrupt communication with other devices.  The 0x03 is not a correct address and it will disappear after REM restarts or the detection is re-run.  So the address you need is 63 in decimal 0x3F is a hexadecimal address.  So when the address screen pops up enter 63.

**Comment by rstrouse** [comment]
So to question 1.  The reason this worked differently this time is because I believe you missed one of the commands to install nodejs using the nvm (Node Version Manager).  That is the command `nvm install 12.18.3`.  There are a several of methods to get node and npm (Node Process Manager).  The command that @tagyoureit had you run simply installs it from another packaged source.

If you look at this link you will see both methods of install.
 https://linuxize.com/post/how-to-install-node-js-on-raspberry-pi/

**Comment by rock-crusher** [comment]
I cant get the address screen to open. I've rebooted etc.
Can you help me get it open?

**Comment by rstrouse** [comment]
Hmmm... just to be sure you are clicking on the + symbol on the `I2c Bus #1` tab.
![image](https://user-images.githubusercontent.com/47839015/143328100-90eed318-4af7-4f19-9f6f-a2d11bbeef2f.png)

If so what browser are you using?

**Comment by rock-crusher** [comment]
yes

**Comment by rstrouse** [comment]
Which browser?  Chrome, Edge, or something else?

**Comment by rock-crusher** [comment]
chrome

**Comment by rstrouse** [comment]
Ok so I open run `pm2 monit` and watch the console when you click on the plus symbol.  It is very odd that your addresses in i2cdetect start at 8 not 3.

**Comment by rstrouse** [comment]
Shoot me a screenshot of your interfaces tab on the Raspberry Pi Configuration.  It should have only I2C and Serial Port enabled.  All others should be set to Disable.

**Comment by rock-crusher** [comment]
I will disable SPI
![e](https://user-images.githubusercontent.com/94593954/143329860-96c80174-f51d-4271-a19b-b18ea55cbe9d.png)

**Comment by rock-crusher** [comment]
This is screenshot 

> Ok so I open run `pm2 monit` and watch the console when you click on the plus symbol. It is very odd that your addresses in i2cdetect start at 8 not 3.

This is before disabling SPI

![d](https://user-images.githubusercontent.com/94593954/143330058-d17cf505-4673-4deb-85af-5a79a02fc471.png)

**Comment by rstrouse** [comment]
That tells me that the I2c bus is not communicating.  If you stretch that screen to the right you will get more info.  What hardware is installed on your pi?

**Comment by rock-crusher** [comment]
just the building automation card & relay card if that is what you mean.

**Comment by rstrouse** [comment]
Run the following commands
`pm2 stop REM`
`cd ~/relayEquipmentManager`
`npm start`

This way we can watch the console when you run.

**Comment by rock-crusher** [comment]
now what?

**Comment by rstrouse** [comment]
What is the complete error for the BAS trying to communicate?

**Comment by rstrouse** [comment]
Also when you ran npm start did any error messages appear in the console?

**Comment by rock-crusher** [comment]
> Run the following commands `pm2 stop REM` `cd ~/relayEquipmentManager` `npm start`
> 
> This way we can watch the console when you run.

![g](https://user-images.githubusercontent.com/94593954/143334106-f5acfd26-bfc1-4b9d-80f6-36c486ecba89.png)

**Comment by rock-crusher** [comment]
Hope this is what you were asking for?

**Comment by rstrouse** [comment]
Ok so it looks like you have more than one REM server loaded.   Did you run `pm2 stop REM`?

**Comment by rock-crusher** [comment]
yes

**Comment by rock-crusher** [comment]
do it again?

**Comment by rstrouse** [comment]
Yes this time it should show that REM is stopped.  Do you have it running in any other window?

**Comment by rock-crusher** [comment]
No, I'll do it again.

**Comment by rock-crusher** [comment]
Wait, I do have it open now, but in one window is that ok?

**Comment by rstrouse** [comment]
If you mean on the pi then no that is not ok.  When we run `npm start` we are starting another copy and there can only be 1.

**Comment by rock-crusher** [comment]
Here is my screen, is this ok?
![h](https://user-images.githubusercontent.com/94593954/143341353-1e7a593c-2890-4b73-9e7e-4436bf8a261b.png)

**Comment by rstrouse** [comment]
I am talking about terminal windows.  The web page is ok as this is not a REM server.  Only the program launched by `npm start` or a pm2 process is a server.  You didn't also set up systemd from the same Wiki page did you?

**Comment by rstrouse** [comment]
The reason why I am saying that REM is still running is because of this message on the npm start.  That indicates that the server is already running on the designated port when we are trying to start it.
![image](https://user-images.githubusercontent.com/47839015/143347481-685d3e1f-dd8a-45ac-9e7b-a814d34441d2.png)

**Comment by rock-crusher** [comment]
No, the instructions never said it install that.

**Comment by rstrouse** [comment]
So what does the prompt say when you run `pm2 stop REM`

**Comment by rock-crusher** [comment]
I ran it again

![j](https://user-images.githubusercontent.com/94593954/143349797-4e4a766f-01d4-4194-befb-727cdd972d55.png)


> So what does the prompt say when you run `pm2 stop REM`

![i](https://user-images.githubusercontent.com/94593954/143350709-1d853631-def1-43bd-850e-3654471c3800.png)

**Comment by rstrouse** [comment]
Ok that is much better.  It also appears to be now communicating with the MEGA-BAS correctly.  Keep this console open and capture what goes by when you click the + on the devices list.

**Comment by rock-crusher** [comment]
It gave me that same wrong address?

![l](https://user-images.githubusercontent.com/94593954/143358327-d438454e-15e3-490d-bf59-7e9b24330f70.png)

![k](https://user-images.githubusercontent.com/94593954/143358528-eab8ff5a-2b47-4023-abe6-c82592927172.png)

**Comment by rstrouse** [comment]
Wait did you enter 63 and it still gave you 3?

**Comment by rstrouse** [comment]
Not 3F you must enter it in Decimal which is 63.

**Comment by rock-crusher** [comment]
I didn't enter anything, the window to do that never opened, the program just does it by itself.
This has been the problem from the beginning.

**Comment by rstrouse** [comment]
Open up a browser on your windows and go to address `http://raspberrypi:8080`.  Could it be that the click is being propagated to the window that opens up.

**Comment by rock-crusher** [comment]
It shows the same thing.

**Comment by rstrouse** [comment]
Ok so close the console that we currently have running rem by clicking the x at the upper right.

Then
`cd ~/relayEquipmentManager`
`git pull`
`npm i`
`npm start`

Wait for REM to start up.

Then using the browser on your PC hard refresh it by holding down the ctrl key and pressing F5.

I forcibly removed the mouse click event from the queue when you click the +.

**Comment by rock-crusher** [comment]
Ok, should I try again clicking the device + button?

**Comment by rock-crusher** [comment]
I clicked it again, sorry. It did the same thing.

**Comment by rstrouse** [comment]
Ok that is bizarre.  From the `~/relayEquipmentManager/data` directory upload your controllerConfig.json file.  I will add the relay in manually for you.  I cannot believe this is acting this way.

**Comment by rock-crusher** [comment]
Sorry, had to step away for awhile.
The system (this site) says: "We don't support that file type."
How can I upload it?

**Comment by rstrouse** [comment]
Rename it to .txt

**Comment by rock-crusher** [comment]
[controllerConfigjson.txt](https://github.com/tagyoureit/nodejs-poolController/files/7600170/controllerConfigjson.txt)

**Comment by rock-crusher** [comment]
Here is something else weird, nothing shows under device info?

![m](https://user-images.githubusercontent.com/94593954/143381115-3c6601d7-1efe-4822-9bf4-63f7a919525f.png)

**Comment by rock-crusher** [comment]
I just did a reboot & it now gives a reading?

![n](https://user-images.githubusercontent.com/94593954/143383281-e4722705-49f9-414a-829c-2287741933da.png)

**Comment by rstrouse** [comment]
That is really strange.  I have not seen nor can I explain the weird behavior on your pi.

At any rate here is the file with the pre-configured 8-relay.  
1. Shut down REM.  Make sure there are no instances of REM running `pm2 stop REM`
2. Rename thise file to controllerConfig.json and copy it into the `~/relayEquipmentManager/data` directory
[controllerConfigjson.txt](https://github.com/tagyoureit/nodejs-poolController/files/7604099/controllerConfigjson.txt)
3. Restart REM `pm2 restart REM`

On a side note I noticed two additional i2c bus definitions detected on your pi (Bus 20 & 21).  Both of them are tied to Broadcom STB (Set Top Box) which seem to point to some sort of display device (Roku, XBMC, game console ...etc).  I have just never seen these before and do not normally exist on Raspberry Pi.

Just curious what is the output from `i2cdetect -l`?  Something just doesn't seem right with the pi.

**Comment by rock-crusher** [comment]
Hi rstrouse,
HAPPY THANKSGIVING!
I'll try the above later and let you know how it works out.
Thanks again for all your hard work, I really appreciate it!

**Comment by rstrouse** [comment]
Happy Thanksgiving!  I this works I know what I will be thankful for.   I did order a few more SD cards to run through a full install.

**Comment by rock-crusher** [comment]
Hi rstrouse,
Well, the above worked, I now have the relay board connected and working (thank you for that)!
Also, I am able to save the port "/dev/ttyS0" where before it wouldn't let me.
Problem now is, my VSF pump still does not acknowledge a connection using the RS485 cable. 

![b](https://user-images.githubusercontent.com/94593954/143618788-b6c63616-b16b-4dcd-aaa7-975647580096.png)
![c](https://user-images.githubusercontent.com/94593954/143618802-e51fbed3-e9b1-4f49-ae28-c2d566e1e25a.png)
![d](https://user-images.githubusercontent.com/94593954/143618820-934187bf-c5f8-4a4e-acf1-71de84e6061d.png)
![a](https://user-images.githubusercontent.com/94593954/143618828-cc0cfb8b-d981-4291-b49b-aa4904456157.png)

**Comment by tagyoureit** [comment]
Glad you got it working so far!

Did you add any hardware/plug-ins/etc that would add 2 extra i2c busses?  Those aren't standard?

Also, how are you running your VSF pump today?  How do you turn it on/off and/or does it turn itself on automatically or on a schedule?

**Comment by rock-crusher** [comment]
> Glad you got it working so far!
> 
> Did you add any hardware/plug-ins/etc that would add 2 extra i2c busses? Those aren't standard.
>
> Also, how are you running your VSF pump today? How do you turn it on/off and/or does it turn itself on automatically or on a schedule?

I do have the jumpers set for (2) temp sensors on the Building Automation hat , but they are not connected. Other than that, just the 2 wire RS485 cable to pump.

![f](https://user-images.githubusercontent.com/94593954/143620950-1aa98641-7af1-441f-a011-46efb4d94779.png)
I also have the jumpers set for the RS485
![g](https://user-images.githubusercontent.com/94593954/143621504-1353ad1d-46d4-45da-87c7-21c33f419d8e.png)
Pump configuration 

![e](https://user-images.githubusercontent.com/94593954/143620206-0b682777-5792-4dfd-9607-eaf30488445f.png)


I can only turn the pump on/off at the pump itself, no schedule is set at the pump.

**Comment by tagyoureit** [comment]
What is showing on the pump display when you try to control it?  It should be idle.  Or you can try to turn of Feature 1 and then try to control the pump.  

You could always try to reverse the wires again; some adapters are wired differently/wrong.  Also, send a couple pictures of the wiring to your Pi and the pump.

**Comment by rock-crusher** [comment]
> What is showing on the pump display when you try to control it? It should be idle. Or you can try to turn of Feature 1 and then try to control the pump.
> 
> You could always try to reverse the wires again; some adapters are wired differently/wrong. Also, send a couple pictures of the wiring to your Pi and the pump.
> 
> Oh - wait... why do you have the pump setup as "spa"? Can you set that to pool and then turn on the pool circuit. Spa will try to rotate valves and let's start with something basic like the pool.

Ok, reversed the wires, no change, so I put them back as : Yellow = A, Green = B.
I also changed from "spa" to pool, still nothing.
I'm bench testing everything first prior to putting into service.
I have keyboard, mouse & monitor connected to pi as well.
![h](https://user-images.githubusercontent.com/94593954/143627357-f139e365-3ecd-43cc-931b-1e984c4394eb.jpg)
![i](https://user-images.githubusercontent.com/94593954/143627377-5843baeb-d147-49f4-829f-9f3f3d4f489e.jpg)
![j](https://user-images.githubusercontent.com/94593954/143627386-e7d6f929-be34-49ed-90b3-cfd519938444.png)

**Comment by tagyoureit** [comment]
The pump IS running on a schedule.
![image](https://user-images.githubusercontent.com/1646840/143633091-0a6df615-02e3-4756-b1e0-840890df7734.png)

It won't respond to the RS485 when that is happening.  For starters, press the 1 button to manually turn it on and then try to connect to it with njsPC.  Long term, you'll need to disable the local pump schedule.

**Comment by rock-crusher** [comment]
> The pump IS running on a schedule. ![image](https://user-images.githubusercontent.com/1646840/143633091-0a6df615-02e3-4756-b1e0-840890df7734.png)
> 
> Long term, you'll need to disable the local pump schedule.
I went through the menu, but didn't see how to disable that?

**Comment by tagyoureit** [comment]
Here's the link to the manual: https://www.pentair.com/content/dam/extranet/nam/pentair-pool/residential/manuals/pool-pumps/intelliflo-vsf/intelliflo-vsf-om-eng.pdf

**Comment by rock-crusher** [comment]
Ok, went through the manual and read about programming/scheduling the pump etc.
Nothing has changed, and there is not a setting that will make the "Running 
Schedule" (what you have circled in red) go away other than pressing the start/stop button. 
There are no scheduled programs set at all in my pump, NONE.
All the programs in my pump are either set to disabled, or manual.
The only options for programs 1-4 are: Manual, Schedule, or Egg Timer. Mine has always been set to manual for 1-4.
Now, as for programs 5-8 there are just 2 options: disabled, or schedule. Mine has always been set to disabled.
The "Running Schedule" that shows up on the screen seems to always be there no matter what settings are in place.
Again, the only way I have found to make that go away is to press the start/stop button.
So, maybe there is something wrong with my pump/display, or?
I must say though, I'm a little confused about this (what else is new) because when I asked a few days ago, rstrouse said the pump needs no special set-up on its control panel in order to connect with the Nixie pool controller, that it was just plug & play. 
Thoughts?
![k](https://user-images.githubusercontent.com/94593954/143672612-230ada02-364d-4fe0-9931-67598c28e6d6.png)

**Comment by rstrouse** [comment]
Well we are not done here.  I have a couple of SD cards that are arriving today so I can rework the setup just as you have it.  

It is my belief that the serial port is not really communicating.  The serial port on the Raspberry Pi through njsPC is putting outgoing messages onto the RS485 bus but I do not believe they are going anywhere.  Hence the reason why your pump that appears to be properly set up is not responding.  

There is one other thing on the pump to check and that is to make sure the pump is on address 1.  Follow the docs for the pump and ensure address 1.  Externally from the pi this is the one thing that would send the data out on the RS485 bus and the pump won't hear it.

If checking the address doesn't get us there, lets go back to the basics.  First, it is alarming that there is an i2c-20 and i2c-21 on your Pi.  These buses are non-existent in the Raspberry Pi docs.  In fact I don't see a way to enable them or to map pins to them.  Since this is the only thing that I see being weird about your pi lets get to the bottom of it.

This is why we keep asking what hardware is installed.  So lets try to get to the bottom of what these devices are and which RPI pins are being occupied by them.  So lets see what the pins on your pi are mapped to by running the following command and posting the results here.
`raspi-gpio get`

Then power off the pi and remove the relay card.  Power the pi back up again and see if the i2c by running
`i2cdetect -l`
We are trying to see if the i2c-20 and i2c-21 buses disappear.  If they do stop here and let me know.

If they do not disappear then power down the Pi and remove the MEGA-BAS card.  You will have to power the pi through the 5v USB-C connector but power it back up that way and lets see if the i2c-20 and i2c-21 buses go away.

If it does leave your pi running like this and report back here.

**Comment by rock-crusher** [comment]
Yes, my pump is set to address 1, I believe that is its default setting same as with the Nixie under the pumps tab configuration.
Ok, first two pics are: raspi-gpio get
Third pic: i2cdetect -l with relay card removed
Forth pic: MEGA-BAS card removed
Looks like no change?

![a](https://user-images.githubusercontent.com/94593954/143719109-ffecb7d8-4f2a-4521-9012-412268dba419.png)
![b](https://user-images.githubusercontent.com/94593954/143719117-188f4742-c52b-44dd-8551-50c5708a692c.png)
![c](https://user-images.githubusercontent.com/94593954/143719127-5dc37882-1522-4b70-ae90-b35fca9a9994.png)
![d](https://user-images.githubusercontent.com/94593954/143719132-1b6dc343-0aef-4d66-9473-6bbcd07c485c.png)

**Comment by rstrouse** [comment]
Well the good news is that neither of those buses are assigned to any of the IO pins on the pi.  The bad news is that I do not know what they are.

Lets make sure we are up to date on the firmware for Raspberry Pi.  If it asks for confirmation always answer yes at the prompts.
`sudo apt update`
`sudo apt upgrade`

**Comment by rock-crusher** [comment]
Ok, here are the results.

![e](https://user-images.githubusercontent.com/94593954/143721164-f087c597-1d2d-4cbd-b9af-2bff48cce222.png)

**Comment by rstrouse** [comment]
I should have my SD cards in the next couple of hours but I want you to try it with the 1 button pressed on your pump.
![image](https://user-images.githubusercontent.com/47839015/143721738-51bb4489-59a5-4cee-b86f-aa275036d456.png)

You have bullseye installed and I don't think we have seen anyone with that version of the pi OS so I will install a pi with Bullseye and see if there are compatibility issues with that OS.  Some hat manufacturers have had some issues.

**Comment by rock-crusher** [comment]
Only thing that happens when the 1 button is pressed on the pump is that it turns on to the set speed of 450 rpm. Still no control with the Nixie.

**Comment by rstrouse** [comment]
I have installed the new OS and I found that it creates two i2c buses at 20 and 21 and it omits addresses 3-7 on its scan.  That being said this OS is an exercise in futility.  I think it has some very serious warts.  

After fighting with trying to get XRDP installed and realizing that remote desktop is so slow it borders on unusable I will continue to try to see if there are issues related to the serial port on the MEGA-BAS.  I'll let you know.  I am also going to image a Buster for comparison.

**Comment by rstrouse** [comment]
Ok so I have been around the block with the new OS and I found a few things.

First, I would like you to verify a few things.  So **power everything down**.  This includes the pump as well as the Raspberry Pi.
1. Make sure you have the yellow wire in the A terminal and the green wire in the B terminal.   I have several adapters on my desk and many of them are marked incorrectly.  I have modified the docs but it is very likely that at the time of the writing, the adapter I was using was marked incorrectly.
2. After you have made the connections make sure engage the wire into the connector.   I spent about an hour trying to figure out why I only had reliable one way communication.  Turns out you can get the wire into the connector but it ends up not being engaged in the connector.  So screw the connector open quite a bit before inserting the wire and pull on it to make sure it is engaged.
4. Double check the jumpers on the MEGA-BAS to ensure they are installed vertically.  If the jumpers are horizontal that is incorrect and the comms will just loop back on themselves.

![image](https://user-images.githubusercontent.com/47839015/143781526-4765403c-adfa-4e8a-802c-bb0eaa93731f.png)

5. Start the pi back up again and run the following commands.  We want to shut njsPC and REM down.  Make sure that the display indicates that both of these are stopped.
`npm stop njsPC`
`npm stop REM`
6. We need to update the firmware on your MEGA-BAS so run the following commands but only after you have completely turned off REM and njsPC.  This is important as the MEGA-BAS will silently fail the firmware update if it is getting commands.
`cd ~`
`git clone https://github.com/SequentMicrosystems/megabas-rpi.git`
`cd ~/megabas-rpi`
`sudo make install`
`cd ~/megabas-rpi/update 0`
`.update/0`
7. We need to disable the rs485 MODBUS functions for the card so run the following commands.
`cd ~/megabas-rpi`
`megabas 0 rs485wr 0 9600 1 0 1`
8.  The new OS requires that the serial port code is recompiled in njsPC.  So to do this enter the following commands.
`cd ~/nodejs-poolController`
`npm rebuild`
9.  Reboot your pi by entering the following command.
`sudo reboot now`
10.  Wait until the pi has had some time to boot back up again and the njsPC, dashPanel, and REM have had time to start then check the pump display.  What we are hoping for is that the pump says Display Not Active.  If it does not then we need to check the configuration in dashPanel to ensure it is enabled and sending messages to the pump.

**Comment by rock-crusher** [comment]
Hi again rstrouse,
I removed the RS485 wires from the terminals and reinstalled them as stated above, they are tight.
Double checked the jumpers as stated above, they are also correct and in place.
When I try to shut njsPC and REM down I get this:
I have rebooted twice along with a complete shut down & reboot with the same result???

![a28](https://user-images.githubusercontent.com/94593954/143783212-ff72bc09-c8d8-4594-bb49-d08741a319b1.png)
![b28](https://user-images.githubusercontent.com/94593954/143783220-71a4d859-d30f-436a-964e-2c94eaa7f677.png)

**Comment by rstrouse** [comment]
I'm such an idiot.  I modified the post it should be `pm2 stop njsPC` and `pm2 stop REM`

**Comment by rock-crusher** [comment]
Lol... "I'm such an idiot" No sir, far from the truth! 
Hey, That statement is reserved for me! Sorry, you can't use it.
Ok, I'll give it another go and report back.

**Comment by rock-crusher** [comment]
I don't think this turned out correctly?

![c28](https://user-images.githubusercontent.com/94593954/143784336-dd21efaf-ccb4-4122-82ad-cf53bb40ebc8.png)

**Comment by rstrouse** [comment]
It should be `./update 0` whose fingers have I been using.

**Comment by rock-crusher** [comment]
Now this?

![d28](https://user-images.githubusercontent.com/94593954/143784570-67410f0f-70ea-484a-96da-13a5af6e907a.png)

**Comment by rstrouse** [comment]
Ok that looks like REM or njsPC are running.  Run `pm2 stop njsPC` and `pm2 stop REM` then try again.

**Comment by rstrouse** [comment]
Btw you can pick up where you left off.  Just make sure njsPC and REM are not currently running.

**Comment by rock-crusher** [comment]
Ok, all that went good!
Now this.

![e28](https://user-images.githubusercontent.com/94593954/143785006-263cdef0-dddc-405d-9399-4170c8e3923e.png)

**Comment by rstrouse** [comment]
In that directory run `megabas -list`

**Comment by rock-crusher** [comment]
![f28](https://user-images.githubusercontent.com/94593954/143785264-b68173bd-b7ea-4009-8668-87c924a49ea9.png)

**Comment by rstrouse** [comment]
Power down the pi and restart it.

After it boots run the following commands.
`pm2 stop njsPC`
`pm2 stop REM`
`cd ~/megabas-rpi`
`git pull`
`sudo npm make install`
`cd ~/megabas-rpi/update`
`./update 0`
`cd ~/megabas-rpi`
`megabas 0 rs485wr 0 9600 1 0 1`

If all this works continue on with number 8 above.

**Comment by rstrouse** [comment]
I updated the call above for `sudo make install`

**Comment by rock-crusher** [comment]
![f29](https://user-images.githubusercontent.com/94593954/143786103-f9b49b01-1416-44ed-ba86-fbaa60d39215.jpeg)
You are the man!!!
Works beautifully!
Can't thank you enough!
I'm going to play around with this now, should be loads of fun!
Question, How can I go about saving all this (everything on the sd card) to my computer in case something goes wrong. 
I wouldn't want to go through all this again if I didn't have to.

**Comment by rstrouse** [comment]
Well Ahab got his whale this time but only after getting tangled up in the harpoon line!

You can use the SD card copier that comes with the Raspberry Pi OS or create an image file on a USB drive.  Here are some instructions.
https://www.tomshardware.com/how-to/back-up-raspberry-pi-as-disk-image

**Comment by rock-crusher** [comment]
Well, I've  been having a good time playing around with this controller setting up circuits, valves, circuit groups, schedules etc, but I do have some questions.

1. I cant seem to get the heater to turn on with the pool, or spa. I have the heater set-up and assigned to a relay and also that same relay enabled in REM. Shouldn't the heater automatically turn on when the spa is activated?

2. Is there a way to start a scheduled event without having to wait until the next day when it's scheduled to turn back on? For example, by simply clicking an event button the timed schedule starts and runs to the end?

3. Is there a simple way to create several timing macros and have them be executed by clicking an event button? For example, the event button is clicked and the timed events are initiated in sequential order until timed out.

4. Intellibrite lights, how do you set-up the relay commands for changing the colors etc? I have the Intellibrite light set-up in the Light Group with the color set as Cyan. It shows up in the lights section of the dash panel, but when it is turned on
the relay only clicks on once and that is it, it doesn't send the appropriate on/off/on sequences to the relay in order to control the light for the Cyan color. It doesn't matter which color you chose, the relay only switches on once, no cycling
on/off of the relay.

Thanks again for all your help!

**Comment by rstrouse** [comment]
1.  Do you have a Water Temperature feed to Nixie.  If you do not then it will not turn on the heater.   Nixie will act as a Thermostat so when the temperature falls below the requested setpoint it will turn on the heater when the temperature is reached it will shut the heater off.

2.  There are 2 types of schedules.  If you create a schedule for a circuit or feature it will automatically turn on at the time of the schedule.  If you manually override it by turning said circuit off then it will go off.  However, if you turn it back on within the scheduled period it will then finish out the schedule.  Then there are egg timers.  Egg timers are set up so that when a circuit or feature is manually engaged it will only run for the amount of time on the timer.  This is very handy for things that you don't want to be left on.

3. You do not execute schedules you execute the circuit.  The schedules take care of themselves.

4. IntelliBrite lights are baked into the circuit relays.  So if you set the circuit type to IntelliBrite it will allow you to change the theme.   Light groups operate differently, these are a collection of individual lighting circuits.  So if you are expecting to set Cyan using the sAM colors tab you may be a bit disappointed.  If flat out sucks.  It is a crap shoot in IntelliTouch and IntelliCenter whether you actually get the sequence to work every time.  At some point I am going to re-sequence these lights manually.  However, the Themes should work just fine on the individual lights.

**Comment by rock-crusher** [comment]
Hi rstrouse,
Thank you once again for the clarification.
Sorry if I'm not using the correct lingo when trying to ask my questions.

Ok, for question 1, no, I do not have temp sensors connected (makes sense) so that explains that.
I thought maybe the relay would kick on and then it would rely on the actual heater sensors/control panel to determine if it would fire-up or not. Thinking about it, I guess it could be set-up in a roundabout way like that, but why would you want to, right?

You know, my heater (MasterTemp 400) has the RS485 connection in its circuit board. I understand this is new just this year 2021. Will this controller be able to accommodate the RS485 connection for heater control rather than using the fireman's switch? 

My take away from your answer to my questions 2 & 3 are no, it cant be done.
I would just like to be able to start one of my schedules at random by simply clicking one button and watch the entire sequence of timed scheduled events take place.
Apparently this is not possible, ie; you must wait 24hrs for the schedule to restart on its own.
I do however understand the egg timers, but these alone wont achieve what i'm looking to do.

> 4. IntelliBrite lights are baked into the circuit relays.  So if you set the circuit type to IntelliBrite it will allow you to change the theme.  

Yes, I have done this, thanks, I just didn't understand the Light group part, kept getting errors and didn't seem to work.

**Comment by rstrouse** [comment]
> You know, my heater (MasterTemp 400) has the RS485 connection in its circuit board. I understand this is new just this year 2021. Will this controller be able to accommodate the RS485 connection for heater control rather than using the fireman's switch?

I just added the initial code to control the MasterTemp RS485 for Nixie yesterday evening.  I am waiting on another user to give me some message returns from the heater to finish it off.

> Yes, I have done this, thanks, I just didn't understand the Light group part, kept getting errors and didn't seem to work.

Light groups especially the SAm lighting (the spot where you define a sequence) is really lame.  When Spectrum came up with this protocol the idea was to manage a color wheel.   The color wheel would rotate in front of the light source until the desired color was achieved.   To be fair the switch clicky thing was a remarkable way to achieve some level of control over this since moving a color wheel on a servo was slow anyway.  Unfortunately, this stupid protocol stuck to this day with no possibility of it going away any time soon.

I can't fault Pentair on this one because every major manufacturer implemented their lighting this way and have not given up on the switch clicky thing.  It takes a full minute for the lights to respond using this protocol using only the 3 pool lights in my pool.  Then it is a crap shoot as to whether the lights sequence as I have defined them.  Bear in mind this is not a Nixie thing or an IntelliCenter thing it is baked into the light fixture.

I have some ideas to make this better than the Pentair implementation so when I get time I will have a look.  For now I would not try to do much with the SAm mode lighting.  I have spent hours using IntelliCenter and IntelliTouch to try to create an effect.   Bear in mind I have a two full DMX universes for landscape lighting and I have spent less time sequencing that successfully than failing at IntelliBrite.

If you want to understand where these guys were coming from this old marketing video will give you some perspective.  Unfortunately, IntelliBrite, ColorLogic, and Jandy WaterColors all implement the protocol of the SAm light.  Even after we got better LED performance.  When my 5 IntelliBrites fail I will gut them and pop a DMX decoder in them.  Then they will be able to go to the music like the rest of the yard.  Did I mention that it is lame?
https://www.youtube.com/watch?v=F3SlNbylXnA

**Comment by rock-crusher** [comment]
Hi rstrouse, 
> I just added the initial code to control the MasterTemp RS485 for Nixie yesterday evening. I am waiting on another user to give me some message returns from the heater to finish it off. 

That's awsome, please give a heads-up when completed.

> this old marketing video will give you some perspective. https://www.youtube.com/watch?v=F3SlNbylXnA

Lol... it's funny how they bagged on Hayward for using Led lights, now they are using them too!

Any benefit using the 120v over the 12v version, or vice versa? 
How many 12v lights can be run on a single xformer? 

Another question (sorry), Using an IC40 with this controller the xformer would connect to the "Filters" circuit relay for power, yes?
If this is correct, then when spa is turned on, and/or pool the IC40 will come on as well.
That said, when in spa mode, is there a way for the IC40 to reduce its percentage of output? Via RS485 connection?

Thank you.

**Comment by rstrouse** [comment]
> That's awsome, please give a heads-up when completed.

Will do

> Any benefit using the 120v over the 12v version, or vice versa?
How many 12v lights can be run on a single xformer?

The benefit to 120v lights is simply the elimination of transformers for each light you want to control individually.  That being said, you should look at setting up zones rather than trying to control the them individually given the mediocre control you get with IntelliBrite.   So I finally rewired all mine so that the pool lights are controlled in one zone, spa lights in another and MagicStream in yet another.  Trying to keep the 4 IntelliBrites 5G in the pool in sync drove me insane.  That being said I am glad I pulled 120v lights individually to the pad.

Pentair recommends no more that 3 5G on a single 300W Intermatic Transformer.  Each draws about 70W and this leaves a little headroom for the transformer.

> Another question (sorry), Using an IC40 with this controller the xformer would connect to the "Filters" circuit relay for power, yes?
If this is correct, then when spa is turned on, and/or pool the IC40 will come on as well.
That said, when in spa mode, is there a way for the IC40 to reduce its percentage of output? Via RS485 connection?

All of your assumptions are correct you wire the IC40 transformer to the Filter relay so that it comes on when the pool or spa is turned on.   Nixie supports 1 setpoint for each body defined in the controller so there is a separate setpoint for pool mode and one for spa mode.  The RS485 signal will change IC40 to whichever body is selected.

**Comment by rock-crusher** [comment]
Hi again,
I have been experimenting with the valves and such, but haven't been able to get the "pump off during valve action" to take place. The box is checked for pump off during valve action, but there is no delay in the pump when say the Spa is
turned on. The pump just fires right up with no delay at the same time the valve actuators are turning.
What am I missing here, or isn't that the way it works?

**Comment by rstrouse** [comment]
We have not implemented any delays for Nixie yet.  Unless you have deadhead or hammer positions (which you should not with intake/return) this feature should probably never be turned on.  We will be implementing it though.

**Comment by rock-crusher** [comment]
Ok, I was just trying things out and seeing how everything works.
I actually thought that was a good option to have since it would be easier on the internal valve seal not having to rotate under pressure. That was always my thought anyways. I have new Jandy Never Lube 2"-2-1/2" 3-way CPVC valves with the Intermatic actuators I'll be installing. Are these any good and will they be perfectly OK rotating them while the pump is running at 3,450 rpm?

Is the heater cool-down delay in play?

**Comment by rstrouse** [comment]
> I have new Jandy Never Lube 2"-2-1/2" 3-way CPVC valves with the Intermatic actuators I'll be installing. Are these any good and will they be perfectly OK rotating them while the pump is running at 3,450 rpm?

Those will work just fine and they will be fine for valve rotation at RPM.  The valves move slowly and do not hammer as the rotate although after years of dealing with cracked cams and fishing endpoints I prefer the IntelliValves.  I replaced or fixed every CV24 actuator on my pool at least twice over the years.  Since I replaced all 7 with IntelliValve 3 years ago I haven't had a failure since.

> Is the heater cool-down delay in play?

No it is not yet.  In fact, if you hook your MasterTemp up to RS485 they take away that control but I will know more when I get the output from the other user.  In fact MasterTemp says that it does not require a cooldown delay from the automation.  Any cooldown required is done within the heater.

**Comment by rock-crusher** [comment]
OK, good to know,
Thank you once again!

**Comment by rstrouse** [comment]
I added the valve delays for the pumps.  The default delay is 30 seconds for valve movement.

Pull njsPC and dashPanel.  You will need to do a hard refresh by navigating to dashPanel in the browser then hold down the ctrl key and press F5.

**Comment by rock-crusher** [comment]
Hi rstrouse,
Yes, very cool, works as stated, thanks for implementing it. 
I will most likely use this feature when I finally put everthing into service.
Since you are a very pool/spa savy person, I have another question, but off topic.
I have a broken spa jet nozzle (least of my problems), the one that screws in deep inside the venturi tee with the 9/16" hex head. I cant seem to locate the correct one, I did buy one that looked identicle (Waterway), but no dice, the threads are different.
My pool/spa is 33 years old, so I know I'm in for a challenge, no telling what brand venturi tees were being commonly used back then.
Any suggestions for me?

**Comment by rstrouse** [comment]
I'm not sure I can help with the spa jets.  Waterway makes a couple of different kinds.  I have the Power Storm jets and they require a special Jet Wrench to pull them out.

**Comment by rstrouse** [comment]
I added several delays to the configurable delay manager.  This includes the processing of Heater Cooldown delays.  Pull njsPC and dashPanel and do not forget to hard refresh your browser with the dashPanel page open.

https://github.com/tagyoureit/nodejs-poolController/wiki/Nixie-Controller-Setup#delays

**Comment by rock-crusher** [comment]
Hi rstrouse,
Lol... I did as you suggest below, but now cannot connect to Dashboard?
> Pull njsPC and dashPanel and do not forget to hard refresh your browser with the dashPanel page open.

I went through the process twice, I even re-booted and still get this same message:

![a127](https://user-images.githubusercontent.com/94593954/145109191-6b619040-f3c0-45b4-9f16-1eef91c929f7.png)

**Comment by rstrouse** [comment]
Shut down njsPC and do another pull.  There was a typo in the initialization code for folks that already had a valve delay set up.

**Comment by rock-crusher** [comment]
Well, it connected, but things are screwed up not working correctly.
Don't know how to explain it, but everything looks different now?  
Dash panel is not showing correctly & incomplete.
Circuit groups are missing from the Features list on the dash panel and I cant get them to return.
The dash panel is only showing a partial view, not showing filters, schedules etc.
It wont let me change things either?
What I do wrong this time?

**Comment by rstrouse** [comment]
Right click on the screen and click inspect.  Then click on the console tab to see if there are any errors.  Also shoot a screenshot and upload it here.

**Comment by rock-crusher** [comment]
![b127](https://user-images.githubusercontent.com/94593954/145121134-4d3606df-c847-413c-a163-0df1a554c274.png)

**Comment by rstrouse** [comment]
Pull dashPanel.  I added additional exception handling if the state isn't completed updating.  You must hard refresh your browser after pulling dashPanel,

**Comment by rock-crusher** [comment]
Ok, seems the dash panel  is back to where it was before (the way it looked & how it was organized) but some of the buttons don't work, when clicked on, I get this message:
![c127](https://user-images.githubusercontent.com/94593954/145124216-42449d57-5d81-4bfd-a794-6339dac2ff38.png)

**Comment by rstrouse** [comment]
Click on the Hamburger menu then the backups tab.  Then back up njsPC and REM server.  When it completes it will download a zip file to your browser.  Please upload that file here.

**Comment by rock-crusher** [comment]
[backup.zip](https://github.com/tagyoureit/nodejs-poolController/files/7672362/backup.zip)

**Comment by rock-crusher** [comment]
I can see some of your changes, but what happened to the Heater cool down delay?


![d127](https://user-images.githubusercontent.com/94593954/145125651-bffe6588-19b6-49f3-86e1-1bfceeccdc27.png)

**Comment by rstrouse** [comment]
The heater cooldown delay is part of the heater configuration.  Only gas and MasterTemp have a cooldown delay but Pentair is adamant that it is not required for MasterTemp heaters.  It uses a flue blower to cool down the heat exchanger in a controlled manner.
![image](https://user-images.githubusercontent.com/47839015/145126065-b169718d-17f0-4266-8199-f227b5596bac.png)

**Comment by rock-crusher** [comment]
Ok, thank you for that.
Anything on the zip file?

**Comment by rstrouse** [comment]
Yes it turns out you have declared multiple spa circuits just to set the pump speeds.  That is not right.  Only the pool and/or spa should be set up as pool or spa.

**Comment by rock-crusher** [comment]
It worked like that before, it was the only way to have the delay action working while the valves were moving.
If I didn't do that, there was no delay when the valves operated in any other scenario that I created. The delay was only for pool, or spa.

**Comment by rstrouse** [comment]
Ok so you need to set all those Aux-Circuits that you have set up that are not the pool or spa circuit to generic.

Then delete the following circuits.
return valve
intake valve
heater

Then set up your intake and return valve relays on the valves tab.  This is important as the intake and return valves are controlled by the body select.

On the heaters tab simply set the heater relay there you do not need that set up as a circuit.

What kind of cleaner do you have?  If it is one with a booster pump you need to set up a circuit for the cleaner with a type of Master Cleaner.  In that instance the pool cleaner will not engage until the cleaner booster pump has had time to prime.  These pumps typically need forward pressure to ensure prime.

It really didn't work like that before it was simply ignoring the extra spa and pool circuits.

**Comment by rstrouse** [comment]
For the circuit groups you have I need to finish off the Spa Drain circuit type.  That one scares me as Pentair typically does not allow that to engage anywhere but at the pad.  Spa Drain will cavitate your pump if you leave it on.

**Comment by rstrouse** [comment]
I will make it so the pool and/or spa circuit types cannot be selected for non-body circuits.  Maybe I will just reject saving them that way since Pentair gets weird about it if they are not there.

**Comment by rock-crusher** [comment]
I only created those scenarios in order to try and get the valve delay to work with something other than the pool/spa mode. I found changing circuits from generic to either spa, or pool accomplished this and it worked.
Now, the delay for the valve movement is only for pool/spa.
If you want a delayed valve action for anything other than that it will not work and the pump turns on while the valve is rotating.

**Comment by rstrouse** [comment]
What are you trying to accomplish with the valve delay?  This only exists so you do not take your heated spa water and dump it into your pool.  The reverse is also true so you do not dump cold pool water into your spa.  This is not to save your valve or pump.  The valve cycle and rotation can be done at whatever pressure or speed you have the pump running at.

**Comment by rstrouse** [comment]
Here are the ramblings from Pentair on the subject but this feature is only available on IntelliTouch and IntelliCenter.  You will not find it on other panels such as iAqualink or Hayward.
![image](https://user-images.githubusercontent.com/47839015/145131752-bc8b26c2-0da0-4b2d-86e8-e7621c53b4d5.png)

**Comment by rock-crusher** [comment]
I have always thought it best to not rotate valves under pressure. 
Remember I said my pool was 33 years old, well it was originally built with the Ortega valves. Apparently those are what was mostly used back in the day? My previous pool also had those as well. Anyway, I learned the hard way with those valves, if you rotated them while the pump was running you pretty much destroyed the internal gasket/seal that they employed. I had motorized actuators on the Ortega's and the controller at the time was set-up with the delay feature so anytime the valves were called on to rotate the pump would turn off until the valves completed their rotation. 
Again, I just thought it was a good idea, and was searching for a way to accomplish this with this controller, and I did, but now... it won't allow it.

**Comment by rstrouse** [comment]
Other than the Intake and Return do you have any other valves.  With that delay it will always delay on movement of those valves.  I did not see any other valve defined on your system.

**Comment by rock-crusher** [comment]
No other valves.

**Comment by rstrouse** [comment]
Try setting it up the way I said and you will see what I mean.  It knows when the valve needs to rotate and shuts down the pump for that rotation.  The old Ortega valves (my last pool had them) needed lubed all the time.  The only difference between those and the Jandy are really their need to be lubricated and the seals were not as durable.

Here is your setup doing just that.
![image](https://user-images.githubusercontent.com/47839015/145134015-d1a1bc5b-8a0f-4372-b721-4768574bbd8e.png)

**Comment by rock-crusher** [comment]
Yes, I understand what you are saying.
This is what I had in mind when messing around with the delays:

For example, I want to click on one of my circuit group buttons and have the return valve rotate to spa, then after the delay (30 sec) the pump turns on at 3,450 rpm. This not only brings the water level up to full, but also cleans the surface of the
water by pushing whatever dirt, debris, leaves etc. over and through the spillway. Then, after a short time, that can be turned off and the spa button clicked to fire up the spa with the delays in place, heater firing up etc. just as you say above.

Many times kids will play in the spa when it is not on. They will splash a lot of water out and the level is way down.
Simply turning the spa on without first doing the scenario above will leave the water level low, and what ever dirt/debris the kids brought into it while playing will remain.

Remember I asked about macros? It sure would be nice to have the ability to create such a scenario as the above. 

Scenario: 
Click spa button, return valve rotates to spa, after 30 seconds pump turns on at 3,450 rpm, after 3 min pump turns off intake valve rotates to spa, after 30 seconds pump resumes the 3,450 rpm, after the 10 second heater start delay 
heater fires-up.

**Comment by rstrouse** [comment]
Are you thinking of what is referred to as priming time?  This is when a pump is first engaged.  The priming speed starts for the period of time defined in the Priming time.  This is not typically a feature of VSF pumps but it is a feature of VS pumps.  Part of this has to do with the difference in the drive dynamics for the pump.

There isn't anything to say that Nixie couldn't synthesize such a feature.
![image](https://user-images.githubusercontent.com/47839015/145139792-c8b739d0-8811-4033-9081-cf3852d0a21d.png)

**Comment by rock-crusher** [comment]
No, it's just a sequence of events that take place with time delays in place.

**Comment by rock-crusher** [comment]
Something your home automation would have.

**Comment by rstrouse** [comment]
njsPC supports many HA platforms for features like that.  Our guideline for the Circuit Groups are based on what we see from the major vendor pool controllers at this time.  You do have me thinking about sequences now though.

Here is an example of HA using a Vera plugin that talks to njsPC.  All of the device categories are scene capable.  There is also support for just about any platform out there and support for many of the protocols used by automation platforms.
https://github.com/rstrouse/nodejs-poolController-veraPlugin

**Comment by rock-crusher** [comment]
Well, I'm glad for your epiphany.

I'm happy to see that option is back in play, it's nice to have that flexibility with the delay & valve options.
Only thing, it isn't exactly working as before. The delays are working just fine with whatever valve action you set-up, but the problem is the filter relay will always turn on too. Before the change, the filter relay would ONLY turn on when the spa, or pool button
was clicked. Now, the filter relay always comes on no mater what valve configuration you set-up.

For instance, I click my spa empty button, it is set-up for the intake valve to rotate to spa suction, after the 30 second delay the pump turns on and the spa begins to drain into the pool. Problem is, the filter relay now turns on thus also turning
on the salt cell, it didn't do that before the change. No reason for the filter relay/salt cell to be on while the spa is draining.

There may be a time one might want to change the valve configuration to say a waterfall using the delay feature, but not want the salt cell to run at the same time.

So, I guess my question to you would be: is there a way you can put it back to how it was so the filter relay would only turn on when the pool, or spa was on?

**Comment by rstrouse** [comment]
Hmmm... without the filter relay single speed and dual speed pumps couldn't properly operate.  This was an actual bug.  However, I may be able to do some looking at this and only activate that relay when there is an SS or DS pump involved.  I also just checked in changes that should allow you to activate a spa drain.  Pull njsPC and dashPanel.  This is now way more robust than the Pentair OCPs are.

**Comment by rstrouse** [comment]
Alright I think I have you accommodated.  Pull njsPC and if you didn't earlier pull dashPanel.  Don't forget to hard refresh the dashPanel web page.

So now there is an actual spa drain feature and circuit type.  If you set one of these up it will only engage when there is no body currently running.  The spillway will rotate the valves but if you turn on the spa the spillway will disengage.

This allows valve rotations outside of the pool/spa circuit and maintains proper position for when in pool or spa mode.  The idea here would be so that you do not need a circuit definition for the intake and return relays.  These should manage themselves depending on what happens to be running at the time.  In the mode depicted the spa drain is active.  If I select the pool or spa or spillway it will turn off the spa drain.  It might be wise to set an egg timer on it so that the spa drain only runs long enough to drain down the spa and not cavitate the pump.
![image](https://user-images.githubusercontent.com/47839015/145310999-0554b5a0-3596-44c8-9a2b-74cf0504d2b5.png)

**Comment by rock-crusher** [comment]
Another dumb question, I'm sorry, but I don't remember, to properly pull njsPC and dashPanel do I need to stop them from running first?

**Comment by rstrouse** [comment]
If you are running PM2 using the published ecosystem.config.js file then it will restart on its own.

`cd ~/nodejs-poolController`
`git pull`
`npm i`
`cd ~/nodejs-poolController-dashPanel`
`git pull`
`npm i`

and it should restart on its own.  If you are not running PM2 then restart njsPC and dashPanel by shutting down the process or rebooting.

**Comment by rock-crusher** [comment]
Thank you.
I went ahead and tried it out, I like the heading: "Spa Drain Active" with the flashing skull & crossbones, nice touch!
I still would like to be able to apply delays to the pump anytime there is valve movement.
Can it be done?

**Comment by rock-crusher** [comment]
... without the filter relay turning on.

**Comment by rstrouse** [comment]
I just pushed up more refinement for the pump delays.  It now activates on spillway as well as drain.  I may look at other defined valves down the road but I will reserve those on another setting.   

The purpose of such a delay is so that the heat in the spa is preserved.  Half a minute of cold water into the spa would pretty much cause all to have to evacuate if you switched over to pool mode.  Sure you never move a multi-port valve with pressure on it since there are deadhead positions between the port positions but a typical 2 way or 3 way valve is a another story.  I suppose if you did not have an automation system and didn't move the valves in order it would start draining the spa by the time you got both valves changed but there should never be a condition where any position on the valve deadheads the pump.

You will find that Pentair is the only automation system that offers this capability and it is only for the transition of the intake and return valves.

**Comment by rock-crusher** [comment]
Hi rstrouse,

I'm noticing an odd occurrence with the pump delay when using the spillway feature.

For instance, when the pool is running in pool mode (pump on) and you click on the spillway button the pump pauses and waits the 30 second delay for the valve to rotate, once the delay completes the pump resumes its operation (works as it should).

Now, with the pool still running in spillway mode and you click the spillway button again for it to return back to the pool, the valve rotates back, but the pump continues to run without the delay (the delay is missing).

I was hoping the pump delay for the spillway/spa drain would react the same as does the pool/spa pump delay. 

For example, when the pool is running (pump on) and you click on the spa button the pump pauses and waits the 30 second delay for both valves to rotate into spa mode, once the delay completes the pump resumes its operation and increases the rpm for the spa (works as it should).

Now, with the spa still running at the higher rpm you click the pool button to resume pool operation, the pump pauses and the valves start rotating back to the pool. Once the delay has completed, the pump resumes to the lower rpm set for the pool.
Again, works as it should.

Just wondering why the spillway delay is missing during its return back to the pool?

Hopefully an easy fix?

**Comment by rstrouse** [comment]
I added additional delays for the spillway spillway turn off.  I may need to move this delay for the spillway to its own delay category.  This will be annoying to users that are simply trying to preserve the heat in their spa when switching between pool/spa modes.  The valve movement on Intake/Return should never be an issue while the pump is running at the speeds in which an actuator will turn it and never be in a position to dead head the pump regardless of the position of the valve.

We will see where it lands but for now it should trigger a delay whenever the spillway turns off and another circuit attached to the intake/return valves are on.  It may not be an issue due to how easy it is to cancel a delay on dashPanel though.  If you were running IntelliTouch or EasyTouch the delay cancel is buried in the menus so attempting to get stuff turned on is a nightmare when the delays are turned on and you don't have any indication as to why something is in delay.

Go ahead and pull njsPC as I think I have struck some sort of balance.

**Comment by rock-crusher** [comment]
Seems to be working pretty well, thanks!

**Comment by rstrouse** [comment]
I am closing this so if you have any further issues.  Go ahead and submit a new issue.
